scopes.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Access scopes
+---
+
+diaspora* API features access scopes (or just 'scopes') in order to implement resource access management for API clients.
+
+Scopes are part of [OAuth 2.0 specification][oauth2] which is a base for [OpenID Connect Core 1.0][connect] implemented by diaspora*.
+
+When the API client requests an access token it sends the list of scopes it wants to get access to accoring to the selected authorization flow (see ["Authentication"][authentication]).
+
+Approved scopes may differ from what was requested by the client.
+
+[OpenID Connect Core 1.0][connect] defines a few scopes which are specific for getting the authorized user information. API client must include the `openid` scope to the scope request list. `openid` scope means "allow OpenID Connect authentication" and if missing then API authentication will fail.
+
+diaspora* also supports `profile` and `email` scopes according to [OpenID Connect Core 1.0][connect-scopes].
+
+Other scopes are diaspora* API specific. All scopes besides `openid` are optional.
+
+Each API endpoint requires at least one scope to be approved. If the access token used for making a request wasn't approved for the scope of the respective endpoint then `403` status will be responded.
+
+## Scopes list
+
+### contacts:read
+
+This scope gives access to the following requests:
+- Read aspect list
+- Read aspects membership
+- Read private user profiles of contacts
+- Read contacts of contacts when allowed by contact's aspect setting
+- User search will include contacts who are hidden from public search
+
+### contacts:modify
+
+This scope gives access to the following requests:
+- Add new aspects
+- Rename aspects and change aspects properties
+- Delete aspects
+- Add/remove person to/from aspects
+
+### conversations
+
+This scope gives access to private messaging, i.e. the following requests:
+- Create conversations
+- Send private messages
+
+### email
+
+This scope is defined by the OpenID Connect spec. It allows reading user's own email address using UserInfo route or from ID Token.
```

Comment:
"the UserInfo route"
"from the ID Token"

Do we actually implement this?

Reply:
> Do we actually implement this?

No, just like we don't implement most of the specification I wrote here. But our API is still mostly fiction, so my idea here is to describe what we want to have eventually.